src/amount.cpp

```diff
@@ -15,7 +15,7 @@ CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nBytes_)
     int64_t nSize = int64_t(nBytes_);
 
     if (nSize > 0)
-        nSatoshisPerK = nFeePaid * 1000 / nSize;
+        nSatoshisPerK = nFeePaid * PRECISION_MULTIPLIER / nSize;
```

Comment:
This won't work. `PRECISION_MULTIPLIER` is unsigned (either 32-bit or 64-bit) and cpp will convert all other integers to unsigned 64-bit in this line on 64-bit platforms.